lib/merchant_repository.rb

```diff
@@ -1,22 +1,59 @@
 require 'csv'
 require './lib/sales_engine'
+require './lib/merchant'
 
 class MerchantRepository
-  attr_reader :merchants
-  def initialize(data)
-    # @all = merchants_array
-    @merchants = data[:merchants]
+  attr_reader :all
+
+  def initialize(merchant_path)
+    @all = (
+      merchant_objects = []
+      CSV.foreach(merchant_path, headers: true, header_converters: :symbol) do |row|
+        merchant_objects << Merchant.new(row)
+      end
+      merchant_objects)
+  end
+
+  def find_by_id(id)
+    if (@all.any? do |merchant|
+      merchant.id == id
+    end) == true
+      @all.find do |merchant|
+        merchant.id == id
+      end
+    else
+      nil
+    end
   end
 
-  def merchant(id, merchant_name)
-    x = Merchant.new(merchant[:id], merchant[:name])
-    require "pry"; binding.pry
+  def find_by_name(name)
+    if (@all.any? do |merchant|
+      merchant.name.upcase == name.upcase
+    end) == true
+      @all.find do |merchant|
+        merchant.name.upcase == name.upcase
+      end
+    else
+      nil
+    end
   end
 
-  def merchants_array(merchant)
-    @merchants = []
-    merchants << x
-    # @merchants.map do |merchant|
+  def create(attributes)
+    new_merchant = Merchant.new(attributes)
+    @all << new_merchant
+  end
+
+  def update(id, new_name)
+    if find_by_id(id) != nil
+      (find_by_id(id).name.clear.gsub!("", new_name))
```

Comment:
Hey Philip and Mallory...can either of you help us to understand what line 48 code is?  Thanks